Content.Server/GameObjects/Components/Rotatable/FlippableComponent.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Content.Server.Interfaces;
+using Content.Shared.GameObjects;
+using Content.Shared.GameObjects.EntitySystems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Maths;
+
+namespace Content.Server.GameObjects.Components.Rotatable
+{
+    [RegisterComponent]
+    public class FlippableComponent : Component
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IServerNotifyManager _notifyManager;
+#pragma warning restore 649
+
+        public override string Name => "Flippable";
+
+        private void TryFlip(IEntity user)
+        {
+            if (Owner.TryGetComponent(out ICollidableComponent collidable) &&
+                collidable.Anchored)
+            {
+                _notifyManager.PopupMessage(Owner.Transform.GridPosition, user, Loc.GetString("It's stuck."));
+                return;
+            }
+
+            Owner.Transform.LocalRotation += Angle.FromDegrees(180);
```

Comment:
Adding 180 degrees rotation is not flipping. In SS13 this is used to flip the direction of the arrow on disposals junctions, for example.